framework/db/Connection.php

```diff
@@ -577,12 +577,21 @@ public function open()
         $token = 'Opening DB connection: ' . $this->dsn;
         try {
             Yii::info($token, __METHOD__);
-            Yii::beginProfile($token, __METHOD__);
+            if ($this->enableProfiling) {
+                Yii::beginProfile($token, __METHOD__);
+            }
+
             $this->pdo = $this->createPdoInstance();
             $this->initConnection();
```

Comment:
there is a small chance that someone is using `initConnection` to configure the db connection, i.e. setting `enableProfiling` to another value. To make the code more robust I would get the value `$enableProfiling = $this->enableProfiling` and use the local variable afterwards to keep the nesting of profiling calls.

Reply:
sure.. I've missed that. 
done! 
/cc @cebe @samdark @rob006